From 70e1412cb417eff42abfe171ec1c682dfb51b5f5 Mon Sep 17 00:00:00 2001
From: rbuj <robert.buj@gmail.com>
Date: Wed, 26 Jan 2022 23:01:48 +0100
Subject: [PATCH 4/6] build: fix meson build

---
 src/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/meson.build b/src/meson.build
index eb5a5820..e7dd0784 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -211,7 +211,7 @@ xml_files = [
   ]
 
 foreach xml : xml_files
-  i18n.merge_file(xml,
+  i18n.merge_file(
     input: xml+ '.in',
     output: xml,
     po_dir: '../po',
-- 
2.36.1

